portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,25 +26,48 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> comments = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> micaylaMessages = new ArrayList<>();
-    micaylaMessages.add("My name is Micayla.");
-    micaylaMessages.add("I am from the Bay Area.");
-    micaylaMessages.add("I live in New York City.");
-
     // Converts ArrayList to a JSON string using Gson.
-    String json = convertToJsonUsingGson(micaylaMessages);
+    String json = convertToJsonUsingGson(comments);
 
     // Sends response as JSON.
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  private static String convertToJsonUsingGson(ArrayList<String> messages) {
+  private static String convertToJsonUsingGson(ArrayList<String> comments) {
     Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    String json = gson.toJson(comments);
     return json;
   }
 
-}
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String newComment = getParameter(request, "text-input", "");
+
+    // Add input to ArrayList.
+    comments.add("Comment: " + newComment);
```

Comment:
Consider not adding the "Comment: " here...for a few reasons:

- it's redundant to add it to *every* comment, and while we don't really need to worry too much about memory usage, it is an obvious inefficiency
- the "Comment: " prefix is more of a rendering detail. Imagine if down the road you wanted to change this prefix to "User Response" or even translate it to a different language based on the locality of the user visiting the site. It would be much easier to do that if it's not mixed up with the actual user responses.

I'd suggest you move the addition of "Comment: " into the JS where you render the comments.